packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -0,0 +1,329 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { authentication, FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace externalAppAuthentication {
+  /*********** BEGIN REQUEST TYPE ************/
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type OriginalRequestInfo = IQueryMessageExtensionRequest | IActionExecuteInvokeRequest;
```

Comment:
>OriginalRequestInfo [](http://example.com/codeflow?start=14&length=19)

Is the intent that this type can ever be passed around and actually used without first converting to one of the discriminated union types? E.g., do we want to enable `OriginalRequestInfo` objects to be passed around and code to reason about the `requestType` property without having to first figure out if they have a `IQueryMessageExtensionRequest`  or `IActionExecuteInvokeRequest`? 

As currently defined, that's not a problem. You can pass around `OriginalRequestInfo` objects and get at the `requestType` property without figuring out the real type because both of the types have that property.

However, if the intent is that ALL the types that participate in this discriminated union (even in the future) have a `requestType` property, then that can be enforced by using a base interface instead of a discriminated union, e.g.,
```
interface OriginalRequestInfo {
    requestType: OriginalRequestType;
}
...
interface IQueryMessageExtensionRequest extends OriginalRequestInfo {
...
}
interface IActionExecuteInvokeRequest extends OriginalRequestInfo {
...
}
```